frontend/src/modules/Dashboard/Components/Dashboard.tsx

```diff
@@ -134,7 +145,30 @@ export const Dashboard = () => {
     setSortedOrder(event.target.value as SortOrder)
   }
 
-  const sortedCourses = sorted(courses, sortedOrder)
+  const [currentCourses, setCurrentCourses] = useState<Course[]>(courses)
+
+  const handleRosterChange = (roster: string) => {
+    setCurrentCourses(
+      sorted(
+        courses.filter((course) => course.roster === roster),
+        sortedOrder
+      )
+    )
+  }
+
+  const sortedCourses = sorted(currentCourses, sortedOrder)
+
+  useEffect(() => {
+    const currentSemesterCourses = courses.filter(
+      (course) => course.roster === 'FA22'
+    )
+    setCurrentCourses(currentSemesterCourses)
+  }, [courses])
```

Comment:
An alternative for this would have been to do something like
```js
const [selectedRoster, setSelectedRoster] = useState('FA22')
const sortedCourses = sorted(
  courses.filter(course => course.roster === selectedRoster),
  sortedOrder
)

<MenuItem onClick={() => setSelectedRoster('SU22')}>
```
Curious to hear your thoughts on this. Both ways are definitely acceptable, but this is what I thought of first (it's more similar to how the sorting works with only the selection being kept in state). Yours may be slightly more efficient in that the courses are only filtered on demand.

But I guess if `courses` changes then the user's semester selection will be thrown out since the useEffect will run again. The value of courses might change like when the user modifies courses on the edit page since that state is shared. So it's possible this could cause issues, but I haven't tested and that code hasn't been written yet.

Reply:
Changed so the buttons change the selected roster instead, which also makes the selected roster information available to the rest of the dashboard (for things like CSV). I think I didn't expect the rosters to change when being stored in const and not a state, but I'm assuming it updates since it's based on two other states.

I wasn't exactly sure about the best way to do this either as I know they may want a filtering system in the future. But I'd say this new solution works fine and should still be easy to build the filtering logic on top of it.